File: go/samples/encryption/encryption.go
package main

import (
	"errors"

	"github.com/bloock/bloock-sdk-go/v2/builder"
	"github.com/bloock/bloock-sdk-go/v2/client/entity"
	"github.com/bloock/bloock-sdk-quickstart/utils"
	"github.com/bloock/bloock-sdk-quickstart/utils/logger"
)

func main() {
	utils.Sample("simple encryption/decryption", func(c utils.Config) error {
		payload := "This will be encrypted"
		password := "a STRONG password"

		logger.Info("The following payload will be encrypted: " + payload)

		// To encrypt we add an encrypter to the builder
		encryptedRecord, err := builder.NewRecordBuilderFromString(payload).
			WithEncrypter(entity.NewAesEncrypter(password)).
			Build()

		if err != nil {
			return err
		}

		logger.Success("Encryption successful")

		logger.Success("Encrypted payload: " + string(encryptedRecord.Retrieve()))

		logger.Info("Trying to decrypt with the valid password")

		// To decrypt we build a record from the encrypted record and add a decrypter
		decryptedRecord, err := builder.NewRecordBuilderFromRecord(encryptedRecord).
			WithDecrypter(entity.NewAesDecrypter(password)).
			Build()

		if err != nil {
			return err
		}

		logger.Success("Decryption successful")

		hash, err := decryptedRecord.GetHash()
		if err != nil {
			return err
		}
		logger.Success("Hash: " + hash)

		logger.Success("Decrypted payload: " + string(decryptedRecord.Retrieve()))

		logger.Info("Trying to decrypt with invalid password")

		_, err = builder.NewRecordBuilderFromRecord(encryptedRecord).
			WithDecrypter(entity.NewAesDecrypter("an invalid password")).
			Build()

		if err == nil {
			return errors.New("The password was invalid so an error should've been returned!")
		}

		logger.Success("The password was invalid so the record could not be decrypted")

		return nil
	})
}

File: js/src/samples/encryption/encryption.ts
import { AesDecrypter, AesEncrypter, RecordBuilder } from "@bloock/sdk"
import { Config } from "../../utils/config"
import { Sample } from "../../utils/sample"
import { Logger } from "../../utils/logger";

Sample.run("encryption", async (_: Config) => {
    let payload = "This wil be encrypted";
    let password = "a STRONG password";

    Logger.warn(`The following payload will be encrypted: ${payload}`);

    // To encrypt we add an encrypter to the builder
    let encryptedRecord = await RecordBuilder
        .fromString(payload)
        .withEncrypter(new AesEncrypter(password))
        .build();

    Logger.success(`Encryption successfully`);

    let encryptedPayload = encryptedRecord.retrieve();
    Logger.success(`Encrypted payload: ${new TextDecoder().decode(encryptedPayload)}`);

    Logger.info("Trying to decrypt with the valid password");

    // To decrypt we build a record from the encrypted record and add a decrypter
    let decryptedRecord = await RecordBuilder
        .fromRecord(encryptedRecord)
        .withDecrypter(new AesDecrypter(password))
        .build();

    Logger.success(`Decryption successfully`);

    let hash = await decryptedRecord.getHash();

    Logger.success(`Hash: ${hash}`);

    if (hash != "b6e6816e3c6180fcbda27048f033cf2b6f2a627864240c4c85558bcbece2a2e4") {
        throw new Error("Unexpected hash received");
    }

    let decryptedPayload = decryptedRecord.retrieve();
    Logger.success(`Decrypted payload: ${new TextDecoder().decode(decryptedPayload)}`);

    Logger.info("Trying to decrypt with invalid password");

    let exceptionWasThrown = false;
    try {
        await RecordBuilder
            .fromRecord(encryptedRecord)
            .withDecrypter(new AesDecrypter("an invalid password"))
            .build();
    } catch {
        exceptionWasThrown = true;
    }
    if (!exceptionWasThrown) {
        throw new Error("The password was invalid so an error should've been returned!");
    }

    Logger.success(`The password was invalid so the record could not be decrypted`);
})
